osu.Game.Rulesets.Taiko/UI/TaikoRulesetContainer.cs

```diff
@@ -77,6 +104,25 @@ private void loadBarLines()
             }
         }
 
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            if (!args.Repeat)
+            {
+                var sampleBank = Beatmap.ControlPointInfo.SoundPointAt(WorkingBeatmap.Track.CurrentTime).SampleBank ?? "normal";
+                string sampleName = "";
+
+                if (centreKeys.Contains(args.Key))
+                    sampleName = "hitnormal";
+
+                else if (rimKeys.Contains(args.Key))
+                    sampleName = "hitclap";
+
+                audio.Sample.Get($"Gameplay/{sampleBank}-{sampleName}")?.Play();

```

Comment:
`SampleInfo` should contain a constructor or a static `FromSoundPoint` method to populate itself. It should also contain a `GetChannel(SampleManager manager)` function, which will perform this string interpolation, retrieve a `SampleChannel`, and add its volume to the `SampleChannel`.

In the case that the playfield plays the samples, the sample volume should come from the `SoundPoint`.

`TaikoPlayfield` should load samples during its `load()` for all `SoundPoints`.

Reply:
Where should the sample _names_ be specified though? I mean, `TaikoPlayfield` can load all necessary samples, sure, but which ones actually get played? Do we assume some default value for that based on the fired `TaikoAction`? It seems so, but I'm not sure anymore.

Reply:
Would be fine with specifying/loading the normal and clap samples for each control point in taikoplayfield.

Reply:
And then playing `hitnormal` for centre and `hitclap` for rim hits, right? That's what I'm confused about.. I saw some kind of code where rim was specified as hitclap _or_ hitwhistle, if I remember correctly.